swift/apple/Firezone/Application/FirezoneApp.swift

```diff
@@ -86,6 +91,49 @@ struct FirezoneApp: App {
       maybeShowOutdatedAlert()
     }
 
+    private func enforceSingleInstance() {
+      // Get the actual bundle identifier from the running app
+      guard let bundleId = Bundle.main.bundleIdentifier else { return }
+
+      let runningApps = NSRunningApplication.runningApplications(
+        withBundleIdentifier: bundleId
+      )
+
+      guard runningApps.count > 1 else { return }
+
+      for app in runningApps where app != NSRunningApplication.current {
+        // Try to terminate the other instance
+        var terminated = app.terminate()
+
+        if !terminated {
+          // If graceful termination fails, force terminate
+          terminated = app.forceTerminate()
+        }
+
+        if !terminated {
+          // If we still can't terminate, show an alert to the user
+          DispatchQueue.main.async {
+            let alert = NSAlert()
+            alert.messageText = "Another Firezone Instance Detected"
```

Comment:
```suggestion
            alert.messageText = "Firezone is already running"
```

Maybe that sounds a bit better?

Reply:
I'm not sure, we're already say that "Another instance is already running" just below it... and I like the uppercase & sounding serious with "detected" :)

I guess we can always iterate on it further if it's unclear to any of our users.